src/briefcase/platforms/macOS/__init__.py

```diff
@@ -32,6 +34,35 @@ def run_app(self, app: BaseConfig, **kwargs):
 
         :param app: The config object for the app
         """
+        # Start log stream for the app.
+        # Streaming the system log is... a mess. The system log contains a
+        # *lot* of noise from other processes; even if you filter by
+        # process, there's a lot of macOS-generated noise. It's very
+        # difficult to extract just the "user generated" stdout/err log
+        # messages.
+        #
+        # The following sets up a log stream filter that looks for:
+        #  1. a log sender that matches that app binary; or,
+        #  2. a log sender of libffi, and a process that matches the app binary.
+        # Case (1) works for pre-Python 3.9 static linked binaries.
+        # Case (2) works for Python 3.9+ dynamic linked binaries.
+        sender = os.fsdecode(self.binary_path(app) / "Contents" / "MacOS" / app.formal_name)
+        log_popen = self.subprocess.Popen(
+            [
+                "log",
+                "stream",
+                "--style", "compact",
+                "--predicate",
+                f'senderImagePath=="{sender}"'
+                f' OR (processImagePath=="{sender}"'
+                ' AND senderImagePath=="/usr/lib/libffi.dylib")',
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            bufsize=1,
+        )
+        time.sleep(3)  # let log stream start up
```

Comment:
same here

Reply:
Left a comment about timeout above. I'll comment on `Popen` exceptions here.

I think that catching `Popen` exceptions is probably unnecessary. As I detailed the behavior of `run` in another comment, `run` is a wrapper on `Popen`. In that analysis, it became clear to me that if `Popen` itself raised an exception (sans `TimeoutError`), `run` would let that exception bubble up in to the caller's code. In all the calling code here, we only ever catch `CalledProcessError` for `run` which will only occur for non-zero return codes (and if `check=True`).

So, given this and the fact `Popen` exceptions are discovered at dev time (via invalid arguments) or reveal a deep OS level runtime issue, I think it makes sense not to try to catch any exceptions from `Popen`.

Reply:
I agree it's unlikely to fail; but if it does, we have an opportunity to fail gracefully - in a way that lets the user know it's probably not their fault, and in a way that maximises the likelihood that their bug report will point at the actual cause of the problem. 

Consider - in this case, in the "caught" scenario, the bug report is "Briefcase is raising an error saying it can't start the log stream". In the "uncaught" scenario, the bug report is a `subprocess.CalledProcessError`, and if we're *really* lucky, they'll provide the full stack trace from which we might be able to work back and determine it's the log stream start that has failed. 

Catching the error and reporting the high-level problem is defensive programming on our part.

Reply:
Huh... for some reason I thought Popen generated other errors (akin to `run()`) , but checking the docs, you're right - it's a fairly limited set of errors that would firmly fall in the "what the hell have you done to your computer?!?" edge case. On that basis, what you've got here is probably sufficient; if the user has borked their system enough that they're getting permission errors out of `log`, surfacing the raw error is probably OK.